src/Ledger/Ratify.lagda

```diff
@@ -231,77 +231,82 @@ restrictedDists coins rank dists = dists
         restrict : Credential ⇀ Coin → Credential ⇀ Coin
         restrict dist = topNDRepDist rank dist ∪ᵐˡ mostStakeDRepDist dist coins
 \end{code}
+\begin{code}[hide]
+module _ (Γ : RatifyEnv)(pparams : PParams) where
+
+  open RatifyEnv Γ
+  open PParams pparams
+\end{code}
+
 \begin{figure*}[h!]
+\begin{AgdaAlign}
 {\small
 \begin{code}
-actualVotes : RatifyEnv → CCData → (GovRole × Credential) ⇀ Vote → GovAction → PParams
-            → VDeleg ⇀ Vote
-actualVotes Γ cc votes ga pparams
-  =    mapKeys (credVoter CC) actualCCVotes
-  ∪ᵐˡ  actualPDRepVotes ∪ᵐˡ actualDRepVotes
-  ∪ᵐˡ  actualSPOVotes
-  where
-    open RatifyEnv Γ
-    open PParams pparams
+  activeDReps : ℙ Credential
+  activeDReps = dom $ filterᵐ? (λ x → currentEpoch ≤? (proj₂ x)) dreps
+
+  activeCC : CCData → ℙ Credential
+  activeCC (just (cc , _))  = dom $ filterᵐᵇ (is-just ∘ proj₂) (ccHotKeys ∣ dom cc)
+  activeCC nothing          = ∅
+
+  actualPDRepVotes : GovAction → VDeleg ⇀ Vote
+  actualPDRepVotes NoConfidence  =  ❴ abstainRep , Vote.abstain ❵ᵐ
+                                    ∪ᵐˡ ❴ noConfidenceRep , Vote.yes ❵ᵐ
+  actualPDRepVotes _             =  ❴ abstainRep , Vote.abstain ❵ᵐ
+                                    ∪ᵐˡ ❴ noConfidenceRep , Vote.no ❵ᵐ
+
+  actualVotes : CCData → GovAction → (GovRole × Credential) ⇀ Vote → VDeleg ⇀ Vote
+  actualVotes cc ga votes  =  mapKeys (credVoter CC) (actualCCVotes cc)
+                              ∪ᵐˡ actualPDRepVotes ga
+                              ∪ᵐˡ actualDRepVotes
+                              ∪ᵐˡ actualSPOVotes ga
+    where
+    spos : ℙ VDeleg
+    spos = filterˢ isSPOProp $ dom (StakeDistrs.stakeDistr stakeDistrs)
 
-    roleVotes : GovRole → VDeleg ⇀ Vote
-    roleVotes r = mapKeys (uncurry credVoter) (filterᵐ? ((r ≟_) ∘ proj₁ ∘ proj₁) votes)
+    actualCCVote : Credential → Epoch → Vote
+    actualCCVote c e =
+      case ¿ currentEpoch ≤ e ¿ᵇ , lookupᵐ? ccHotKeys c of
+      λ where  (true , just (just c'))  → maybe id Vote.no $ lookupᵐ? votes (CC , c')
+               _                        → Vote.abstain -- expired, no hot key or resigned
 
-    activeCC activeDReps : ℙ Credential
-    activeCC = case cc of λ where
-      (just (cc , _))  → dom (filterᵐᵇ (is-just ∘ proj₂) (ccHotKeys ∣ dom cc))
-      nothing          → ∅
+    actualCCVotes : CCData → Credential ⇀ Vote
+    actualCCVotes nothing          = ∅ᵐ
+    actualCCVotes (just (cc , q))  =
+      case ¿ ccMinSize ≤ lengthˢ (activeCC $ just (cc , q)) ¿ᵇ of
+      λ where  true   → mapWithKey actualCCVote cc
+               false  → constMap (dom cc) Vote.no
```

Comment:
*Code smell:* `case b of λ where true → ...; false → ...` is equivalent to `if b then ... else ...`

Reply:
You can even use `ifᵈ` here, then you don't need the `¿`

Reply:
Here I was trying to simplify while keeping the code as close to the original as possible (see lines 259---261 of the original code) ...but now I see that it can be simplified further as you suggest.